src/Expander/Monad.hs

```diff
@@ -860,14 +861,14 @@ setTasks = modifyState . set expanderTasks
 clearTasks :: Expand ()
 clearTasks = modifyState $ set expanderTasks []
 
-expandEval :: Eval a -> Expand a
+expandEval :: Core -> Expand Value
 expandEval evalAction = do
```

Comment:
should probably not be called `evalAction` anymore since its type is no longer `Eval a`

Reply:
`expandEval` should probably be renamed too since it no longer takes an `Eval a` as input. but what would be a better name? `expandCore` is the most obvious candidate, but this is not expanding the core, this is evaluating it. `evalCore`? but then what's the difference between `evalCore` and `evaluate`? the difference is that `evalCore` runs in the `Expand` monad. so we are expanding in the current phase. so maybe `evalInCurrentPhase`? that makes sense for something which runs in the `Expand` monad, since only the `Expand` monad has a concept of a current phase.

Reply:
ah great points! `evalInCurrentPhase` seems right to me. I agree that `expandCore` or `evalCore` is too general. I wonder if the naming difficulty is indicative of a larger architectural problem.